fog/view/server/test-utils/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+
+            let ra_client =
+                AttestClient::new(&config.ias_api_key).expect("Could not create IAS client");
+            let mut router_server =
+                FogViewRouterServer::new(config, enclave, ra_client, store_clients, logger.clone());
+            router_server.start();
+            (router_server, db_test_context, store_servers)
+        };
+
+        let grpcio_env = Arc::new(grpcio::EnvBuilder::new().build());
+        let mut mr_signer_verifier =
+            MrSignerVerifier::from(mc_fog_view_enclave_measurement::sigstruct());
+        mr_signer_verifier.allow_hardening_advisory("INTEL-SA-00334");
+
+        let mut verifier = Verifier::default();
+        verifier.mr_signer(mr_signer_verifier).debug(DEBUG_ENCLAVE);
+        let router_client = FogViewRouterGrpcClient::new(router_uri, verifier, grpcio_env, logger);
+
+        Self {
+            db_test_context: Some(db_test_context),
+            router_server: Some(router_server),
+            router_client: Some(router_client),
+            store_servers: Some(store_servers),
+        }
+    }
+
+    /// Creates fog view stores with sane defaults.
+    fn create_view_stores(
+        omap_capacity: u64,
+        store_count: usize,
+        logger: Logger,
+    ) -> (
+        SqlRecoveryDbTestContext,
+        Vec<ViewServer<SgxViewEnclave, AttestClient, SqlRecoveryDb, EpochShardingStrategy>>,
+        Arc<RwLock<HashMap<FogViewStoreUri, Arc<FogViewStoreApiClient>>>>,
+    ) {
+        let db_test_context = SqlRecoveryDbTestContext::new(logger.clone());
+        let mut store_servers = Vec::new();
+        let mut store_clients = HashMap::new();
+
+        for i in 0..store_count {
+            let db = db_test_context.get_db_instance();
+            let (store, store_uri) = {
+                let port = portpicker::pick_unused_port().expect("pick_unused_port");
+                let store_uri = FogViewStoreUri::from_str(&format!(
+                    "insecure-fog-view-store://127.0.0.1:{}",
+                    port
+                ))
+                .unwrap();
+
+                // Each store is responsible for 1 block. Note that this means that the stores
+                // in this test are not responsible for overlapping block ranges.
+                let store_block_range = BlockRange::new(i as u64, (i + 1) as u64);
+                let epoch_sharding_strategy = EpochShardingStrategy::new(store_block_range);
+
+                let config = ViewConfig {
+                    chain_id: "local".to_string(),
+                    client_responder_id: ResponderId::from_str(&store_uri.addr()).unwrap(),
+                    client_listen_uri: Store(store_uri.clone()),
+                    client_auth_token_secret: None,
+                    omap_capacity,
+                    ias_spid: Default::default(),
+                    ias_api_key: Default::default(),
+                    admin_listen_uri: Default::default(),
```

Comment:
:question: If I'm understanding correctly the `FogViewRouterConfig`s `admin_listen_uri` is for admin access _only_ to the router.  This `admin_listen_uri` would be for admin access to _only_ this store instance and thus why the value isn't copied.

Reply:
yea